docs/cinder-csi-plugin/using-cinder-csi-plugin.md

```diff
@@ -41,114 +44,119 @@
+For each kubernetes official release, there is a corresponding release of Cinder CSI driver which is compatible with k8s release. It is recommended to use the corresponding version w.r.t kubernetes.
 
-> NOTE: All following examples need to be used inside instance(s) provisoned by openstack, otherwise the attach action will fail due to fail to find instance ID from given openstack cloud.
+For sidecar version compatibility with kubernetes, please refer [CompatibilityMatrix](https://kubernetes-csi.github.io/docs/sidecar-containers.html) of each sidecar.
 
-### Example local-up-cluster.sh
+## Driver Config
 
-```ALLOW_PRIVILEGED=true RUNTIME_CONFIG="storage.k8s.io/v1=true" LOG_LEVEL=5 hack/local-up-cluster.sh```
+Implementation of `cinder-csi-plugin` relies on following OpenStack services.
 
-### Deploy
+| Service                        | API Version(s) | Deprecated | Required |
+|--------------------------------|----------------|------------|----------|
+| Identity (Keystone)            | v2             | Yes        | No       |
+| Identity (Keystone)            | v3             | No         | Yes      |
+| Compute (Nova)                 | v2             | No         | Yes      |
+| Block Storage (Cinder)         | v3             | No         | Yes      |
 
-You can either use the manifests under `manifests/cinder-csi-plugin` or the Helm chart `charts/cinder-csi-plugin`.
 
-#### Using the Helm chart
+For Driver configuration, parameters must be passed via configuration file specified in `$CLOUD_CONFIG` environment variable.
+The following sections are supported in configuration file.
 
-> NOTE: This chart assumes that the `cloud-config` is found on the host under `/etc/kubernetes/` and that your OpenStack cloud has cert under `/etc/cacert`.
+### Global
+For Cinder CSI Plugin to authenticate with Openstack Keystone, required parameters needs to be passed in [Global] section of the file. For all supported parameters, please refer [Global](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-openstack-cloud-controller-manager.md#global) section.
 
-To install the chart use the following command:
-```
-helm install --namespace kube-system --name cinder-csi ./charts/cinder-csi-plugin
-```
+### Block Storage
+These configuration options pertain to block storage and should appear in the `[BlockStorage]` section of the `$CLOUD_CONFIG` file.
+
+* `node-volume-attach-limit`
+  Optional. To configure maximum volumes that can be attached to the node. Its default value is `256`.
+* `rescan-on-resize`
+  Optional. Set to `true`, to rescan block device and verify its size before expanding the filesystem. Not all hypervizors have a /sys/class/block/XXX/device/rescan location, therefore if you enable this option and your hypervizor doesn't support this, you'll get a warning log on resize event. It is recommended to disable this option in this case. Defaults to `false`
+
+### Metadata
```

Comment:
I think this is generic and should apply to OCCM as well..
so maybe better to doc there then point link to OCCM from here
not a blocking one

Reply:
@jichenjc I have copied from there , its just one parameter , so I thought good to add instead of redirecting

Reply:
ok, as I am not sure it will be updated thus later on it might be inconsistent..
anyway, either way not a big deal